### PR TITLE
Light maintenance

### PR DESCRIPTION
**Pull request by MichaelMure**
This PR is a form of light maintenance on this repo. Details are in each commit, but TL;DR:
- add go module
- add CI
- modernize a bit the code (remove support for go1.0 and go1.2, deprecated functions)
- make sure that the tests pass

### CODE REVIEW

File: .github/test.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Test
+        run: make test-coverage
```
Comment (hiddeco) [inline .github/test.yml]:
```suggestion
        run: make test-coverage

```

File: .github/test.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.18.x, 1.19.x]
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
```
Comment (pjbgf) [inline .github/test.yml]:
```suggestion
        uses: actions/setup-go@v3
```

File: .github/test.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (pjbgf) [inline .github/test.yml]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/test.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+on: [push, pull_request]
+name: Test
+jobs:
+  version-matrix:
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.18.x, 1.19.x]
```
Comment (pjbgf) [inline .github/test.yml]:
```suggestion
        go-version: [1.19.x, 1.20.x]
```

File: go.mod [hunk 5]
```diff
@@ -0,0 +1,8 @@
+module github.com/go-git/gcfg
+
+go 1.12
```
Comment (pjbgf) [inline go.mod]:
```suggestion
go 1.13
```

**Review by pjbgf — CHANGES_REQUESTED** [review]
A few nits to align with versions on `go-git`.

**Review by hiddeco — APPROVED** [review]
Thank you @MichaelMure :bow:

**Comment by MichaelMure** [comment]
BTW, I found out about two maintained fork of `gcfg`:
- https://github.com/please-build/gcfg
- https://github.com/gravwell/gcfg

The first one apparently has a very very nice capability: being able to edit a config file without deleting the comments a user might have put there. It would be great to cherry pick those commits. See for example https://github.com/MichaelMure/git-bug/issues/1022 how that can cause problems.

**Review by pjbgf — APPROVED** [review]
Thank you @MichaelMure!

**Comment by pjbgf** [comment]
@MichaelMure as for the changes in other forks, it would be better to ask the authors whether they want to upstream their changes.

**Pull request merged**